tasks/kmt.py

```diff
@@ -954,6 +956,146 @@ def status(ctx: Context, stack: str | None = None, all=False, ssh_key: str | Non
         print("")
 
 
+@task(
+    help={
+        "version": "The version to update the images to. If not provided, version will not be changed. If 'latest' is provided, the latest version will be used.",
+        "update-only-matching": "Only update the platform info for images that match the given regex",
+    }
+)
+def update_platform_info(ctx: Context, version: str | None = None, update_only_matching: str | None = None):
+    """Generate a JSON file with platform information for all the images
+    found in the KMT S3 bucket.
+    """
+    res = ctx.run(
+        "aws-vault exec sso-staging-engineering -- aws s3 ls --recursive s3://dd-agent-omnibus/kernel-version-testing/rootfs"
+    )
+    if res is None or not res.ok:
```

Comment:
I think if you want to check the error condition explicitly then you need to specify `warn=True` in the call above. Otherwise `ctx.run` will throw an exception.